components/notification/NotificationPaginator.vue

```diff
@@ -112,6 +112,19 @@ function groupItems(items: mastodon.v1.Notification[]): NotificationSlot[] {
   return results
 }
 
+function removeFiltered(items: mastodon.v1.Notification[]): mastodon.v1.Notification[] {
+  const filteredIndices = items.map(
+    (items, index) => (
+      {
+        status: items.status,
+        index,
+      }))
+    .filter(item => !!item.status?.filtered?.find(filter => filter.filter.filterAction === 'hide' && filter.filter.context.includes('notifications')))
+    .map(item => item.index)
+
+  return items.filter((_, index) => !filteredIndices.includes(index))
+}
```

Comment:
Could this be a direct filter with the negated condition you have there? Something like
```
function removeFiltered(items: mastodon.v1.Notification[]): mastodon.v1.Notification[] {
  return items.filter(item => 
    !item.status?.filtered?.find(filter =>
      filter.filter.filterAction === 'hide' && filter.filter.context.includes('notifications')
    )
  )
}
```

Reply:
@patak-dev yeah, that's better. Done!